### PR TITLE
RLP decoding error tracing

### PR DESCRIPTION
**Pull request by MegaRedHand**
Linked to https://github.com/paradigmxyz/reth/issues/325

This PR adds tracing to RLP decoding errors.

It adds two **WARN** level traces: one inside `UnauthedEthStream::handshake` and another one inside `EthStream::poll_next`.

### CODE REVIEW

File: crates/net/eth-wire/src/ethstream.rs [hunk 1]
```diff
@@ -73,7 +73,10 @@ where
 
         let msg = match ProtocolMessage::decode(&mut their_msg.as_ref()) {
             Ok(m) => m,
-            Err(err) => return Err(err.into()),
+            Err(err) => {
+                tracing::warn!("rlp decode error in handshake: msg={their_msg:x}");
```
Comment (Rjected) [inline crates/net/eth-wire/src/ethstream.rs:77]: nit: there are multiple handshakes, so specifying `eth` would make it more clear where the decoding failed
```suggestion
                tracing::warn!("rlp decode error in eth handshake: msg={their_msg:x}");
```

**Review by Rjected — CHANGES_REQUESTED** [review]
awesome tysm! just one nit, then this looks good!

**Review by mattsse — CHANGES_REQUESTED** [review]
thank you for this, would like to see the smol changes @Rjected suggested otherwise lgtm!

File: crates/net/eth-wire/src/ethstream.rs (hunk 1 above)
Comment (MegaRedHand) [reply crates/net/eth-wire/src/ethstream.rs]: Done!

**Review by Rjected — APPROVED** [review]
lgtm!

**Pull request merged**